src/client/app/redux/slices/graphSlice.ts

```diff
@@ -0,0 +1,380 @@
+		changeChartToRender: (state, action: PayloadAction<ChartTypes>) => {
+			state.current.chartToRender = action.payload
+		},
+		toggleAreaNormalization: state => {
+			state.current.areaNormalization = !state.current.areaNormalization
+		},
+		setAreaNormalization: (state, action: PayloadAction<boolean>) => {
+			state.current.areaNormalization = action.payload
+		},
+		toggleShowMinMax: state => {
+			state.current.showMinMax = !state.current.showMinMax
+		},
+		setShowMinMax: (state, action: PayloadAction<boolean>) => {
+			state.current.showMinMax = action.payload
+		},
+		changeBarStacking: state => {
+			state.current.barStacking = !state.current.barStacking
+		},
+		setBarStacking: (state, action: PayloadAction<boolean>) => {
+			state.current.barStacking = action.payload
+		},
+		changeCompareSortingOrder: (state, action: PayloadAction<SortingOrder>) => {
+			state.current.compareSortingOrder = action.payload
+		},
+		updateLineGraphRate: (state, action: PayloadAction<LineGraphRate>) => {
+			state.current.lineGraphRate = action.payload
+		},
+		updateThreeDReadingInterval: (state, action: PayloadAction<ReadingInterval>) => {
+			state.current.threeD.readingInterval = action.payload
+		},
+		updateThreeDMeterOrGroupInfo: (state, action: PayloadAction<{ meterOrGroupID: number | undefined, meterOrGroup: MeterOrGroup }>) => {
+			state.current.threeD.meterOrGroupID = action.payload.meterOrGroupID
+			state.current.threeD.meterOrGroup = action.payload.meterOrGroup
+		},
+		updateThreeDMeterOrGroupID: (state, action: PayloadAction<number>) => {
+			state.current.threeD.meterOrGroupID = action.payload
+		},
+		updateThreeDMeterOrGroup: (state, action: PayloadAction<MeterOrGroup>) => {
+			state.current.threeD.meterOrGroup = action.payload
+		},
+		updateSelectedMetersOrGroups: ({ current }, action: PayloadAction<{ newMetersOrGroups: number[], meta: ActionMeta<SelectOption> }>) => {
+			// This reducer handles the addition and subtraction values for both the meter and group select components.
+			// The 'MeterOrGroup' type is heavily utilized in the reducer and other parts of the code.
+			// Note that this option is binary, if it's not a meter, then it's a group.
+
+			// Destructure payload
+			const { newMetersOrGroups, meta } = action.payload;
+			const cleared = meta.action === 'clear'
+			const valueRemoved = (meta.action === 'pop-value' || meta.action === 'remove-value') && meta.removedValue !== undefined
+			const valueAdded = meta.action === 'select-option' && meta.option
+			let isAMeter = true
+
+			if (cleared) {
+				const clearedMeterOrGroups = meta.removedValues;
+				// A Select has been cleared(all values removed with clear)
+				// use the first index of cleared items to check for meter or group
+				isAMeter = clearedMeterOrGroups[0].meterOrGroup === MeterOrGroup.meters
+				// if a meter clear meters, else clear groups
+				isAMeter ? current.selectedMeters = [] : current.selectedGroups = []
+
+			}
+			if (valueRemoved) {
```

Comment:
Are these mutually exclusive? If so, should it be else if for these?

Reply:
yes, only one event should happpen per reducer pass. elifs were used in the later half of this reducer, but didn't seem to make it up here. Fixed